packages/frontend/src/__tests__/mainPage.js

```diff
@@ -0,0 +1,135 @@
+import UserContext from '../context/User'
+import PostContext from '../context/Post'
+import MainPage from '../pages/mainPage/mainPage'
+
+const mockHistoryPush = jest.fn()
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+        push: mockHistoryPush,
+    }),
+}))
+
+// abstracted render function
+const renderMainPage = (userData, unirepData, postData) => {
+    return render(
+        <UserContext.Provider value={userData}>
+            <UnirepContext.Provider value={unirepData}>
+                <PostContext.Provider value={postData}>
+                    <MainPage />
+                </PostContext.Provider>
+            </UnirepContext.Provider>
+        </UserContext.Provider>
+    )
+}
+
+test('should render MainPage with mocked data and false UserState', () => {
+    // provier data
+    const unirepData = {
+        postReputation: 30,
+        commentReputation: 30,
+    }
+    const userData = {
+        userState: false,
+        netReputation: 30,
+        commentReputation: 30,
+        currentEpochKeys: ['epoch_key test1', 'epoch_key test2'],
+    }
+
+    const postData = {
+        feedsByQuery: {
+            new: {
+                createdAt: '00',
+                content: 'mocked post content',
+            },
+        },
+        loadFeed: jest.fn(),
+    }
+    renderMainPage(userData, unirepData, postData)
+    expect(
+        screen.getByText(/community built on ideas, not identities./i)
+    ).toBeInTheDocument()
+    expect(
+        screen.getByText(/stay up to date & share everything with everyone./i)
+    ).toBeInTheDocument()
+    // userState is *false* so login text is rendered
+    expect(
+        screen.getByText(/you must join or login to create post/i)
+    ).toBeInTheDocument()
+    expect(
+        screen.getByText(/stay up to date & share everything with everyone./i)
+    ).toBeInTheDocument()
+})
+
+test('should render MainPage with mocked data and true UserState', () => {
+    // provier data
+    const unirepData = {
+        postReputation: 30,
+        commentReputation: 30,
+    }
+
+    const userData = {
+        userState: true,
+        netReputation: 30,
+        commentReputation: 30,
+        currentEpochKeys: ['epoch_key test1', 'epoch_key test2'],
+    }
+
+    const postData = {
+        feedsByQuery: {
+            new: {
+                createdAt: '00',
+                content: 'mocked post content',
+            },
+        },
+        loadFeed: jest.fn(),
+    }
+    renderMainPage(userData, unirepData, postData)
+    expect(
+        screen.getByText(/community built on ideas, not identities./i)
+    ).toBeInTheDocument()
+    expect(
+        screen.getByText(/stay up to date & share everything with everyone./i)
+    ).toBeInTheDocument()
+    // userState is *true*
+    expect(screen.getByText(/my rep/i)).toBeInTheDocument()
+    expect(screen.getByText(/30/i)).toBeInTheDocument()
```

Comment:
Would be nice to use `unirepData.postReputation` here (I think that's what this 30 is). We should at least have a comment saying what this number is.